pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -52,19 +83,30 @@ where
 	B::Extrinsic: ExtrinsicCall,
 	<B::Extrinsic as Extrinsic>::Call: IsSubType<crate::Call<PSC>>,
 {
-	let block_data =
-		cumulus_primitives_core::ParachainBlockData::<B>::decode(&mut &params.block_data.0[..])
-			.expect("Invalid parachain block data");
+	let block_data = ParachainBlockData::<B>::decode(&mut &params.block_data.0[..])
+		.expect("Invalid parachain block data");
 
 	let parent_head =
 		B::Header::decode(&mut &params.parent_head.0[..]).expect("Invalid parent head");
 
+	// We decoded the block data and don't need it anymore. So, we release it to free the memory.
+	mem::drop(params.block_data);
+
 	let (header, extrinsics, storage_proof) = block_data.deconstruct();
 
 	let head_data = HeadData(header.encode());
 
 	let block = B::new(header, extrinsics);
-	assert!(parent_head.hash() == *block.header().parent_hash(), "Invalid parent hash",);
+	assert!(parent_head.hash() == *block.header().parent_hash(), "Invalid parent hash");
+
+	let inherent_data = extract_parachain_inherent_data(&block);
```

Comment:
dummy question: should we drop `block.entrinsics` after this?

Reply:
No, we still need to execute them